error.go

```diff
@@ -186,11 +186,11 @@ func raiseIndexOutOfBounds(opts *options, value value, idx int) Error {
 	return raisePathErr(reason, value.meta(), message, ctx.path("."))
 }
 
-func raiseInvalidTopLevelType(v interface{}) Error {
-	// most likely developers fault
+func raiseInvalidTopLevelType(v interface{}, meta *Meta) Error {
+	// could be developers or user fault
 	t := chaseTypePointers(chaseValue(reflect.ValueOf(v)).Type())
 	message := fmt.Sprintf("can not use go type '%v' for merging/unpacking configurations", t)
-	return raiseCritical(ErrTypeMismatch, message)
+	return raiseCritical(ErrTypeMismatch, messageMeta(message, meta))
```

Comment:
use `raiseErr`. `raiseCritical` adds a stack-trace, which we do not need anymore in this case. Reason for error and code-path leading to this error is clear.

Errors generated by ucfg are very likely presented to the user on console. Update the message to be more informative. Something that makes you think... oh I have to have a look at my config file. Something like: `configuration of type '...' is no dictionary or list`.